tests/e2e/network.go

```diff
@@ -0,0 +1,159 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package e2e
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/tests/fixture/e2e"
+
+	"github.com/ava-labs/hypersdk/api/indexer"
+	"github.com/ava-labs/hypersdk/api/jsonrpc"
+	"github.com/ava-labs/hypersdk/chain"
+	"github.com/ava-labs/hypersdk/tests/workload"
+)
+
+var (
+	ErrUnableToConfirmTx = errors.New("unable to confirm transaction")
+	ErrInvalidURI        = errors.New("invalid uri")
+)
+
+const (
+	txCheckInterval = 100 * time.Millisecond
+)
+
+type Network struct {
+	nodes  []*Node
+	parser *parser
+}
+
+func NewNetwork(tc *e2e.GinkgoTestContext) *Network {
+	blockchainID := e2e.GetEnv(tc).GetNetwork().GetSubnet(networkConfig.Name()).Chains[0].ChainID
+	testNetwork := &Network{}
+	for _, uri := range getE2EURIs(tc, blockchainID) {
+		n := &Node{uri: uri, network: testNetwork}
+		testNetwork.nodes = append(testNetwork.nodes, n)
+	}
+	testNetwork.parser = &parser{
+		Parser: networkConfig.Parser(),
```

Comment:
Can we add a comment on the `parser` type explaining why we need to wrap the `chain.Parser` type that is already passed in? As I understand it, this is because we cannot currently know the blockchainID in advance since it's the hash of the transaction on the P-Chain that will create the chain.

Reply:
yes. I'll add some comments around that.